### PR TITLE
Not poisoning restricted types known to be stack only.

### PR DESCRIPTION
**Pull request by VSadov**
Not poisoning restricted types known to be stack only. (as per ECMA 335)

- TypedReference
- ArgIterator
- RuntimeArgumentHandle

Fixes:#22198

===
There are technical reasons for which making RestrictedTypes ref structs is advantageous. In fact it is already done in some cases - see example: https://github.com/dotnet/corert/blob/f23fb52ce14fa680bcebc07eab9a20fa8780e47f/src/ILCompiler.TypeSystem/tests/CoreTestAssembly/Platform.cs#L71

We do not need to poison these types with Obsolete attribute when `ref struct` syntax is used. The stack-only semantics of these types is widely known and poisoning can only lead to compat troubles.

### CODE REVIEW

**Comment by VSadov** [comment]
@dotnet/roslyn-compiler - please take a look

**Comment by VSadov** [comment]
CC @jaredpar - for ask mode approval.

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenMscorlib.cs [hunk 1]
```diff
@@ -373,6 +373,56 @@ public static T Read<T>() where T : new()
 );
         }
 
+        [Fact()]
+        public void NoTypedRefBox2()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenMscorlib.cs]: How does this differ from `NoTypedRefBox1`? Consider adding a comment.

**Comment by VSadov** [comment]
Microsoft.CodeAnalysis.Editor.UnitTests.Structure.StructureTaggerTests.CSharpOutliningTagger_RegionIsNotDefinition     failed windows_debug_unit32_prtest. 

- could not be related to this change.

**Comment by VSadov** [comment]
test windows_debug_unit32_prtest please

**Pull request merged**